code/datums/atmosphere/_atmosphere.dm

```diff
@@ -36,14 +37,13 @@
 	var/datum/gas/gastype
 	var/amount
 	while(gasmix.return_pressure() < target_pressure)
-		if(!prob(restricted_chance))
+		if(!prob(restricted_chance) || !length(spicy_gas))
 			gastype = pick(normal_gases)
 			amount = normal_gases[gastype]
 		else
-			gastype = pick(restricted_gases)
-			amount = restricted_gases[gastype]
-			if(!HAS_TRAIT(SSstation, STATION_TRAIT_UNNATURAL_ATMOSPHERE) && gaslist[gastype])
-				continue
+			gastype = pick(spicy_gas)
+			amount = spicy_gas[gastype]
+			spicy_gas -= gastype //You can only pick each resricted gas once
```

Comment:
```suggestion
			spicy_gas -= gastype //You can only pick each restricted gas once
```

Reply:
Haha I can totally spell, did you know english is my first language?